File: wfm-develop mobile 07.05.2026/wfm/mobile/android/ui/src/main/kotlin/com/beyondviolet/wfm/ui/components/WfmLinkButton.kt
package com.beyondviolet.wfm.ui.components

import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import com.beyondviolet.wfm.ui.theme.WfmSpacing
import com.beyondviolet.wfm.ui.theme.WfmTheme
import com.beyondviolet.wfm.ui.theme.WfmTypography
import com.beyondviolet.wfm.ui.theme.*

/**
 * Link кнопка дизайн-системы WFM
 *
 * Низкий приоритет, для менее важных действий.
 * Представляет собой текст без фона в цветах brand.
 *
 * Дизайн: https://www.figma.com/design/J9crsNS4wIszpILzWvnY2T/Smart-employee?node-id=900-12300
 *
 * @param text Текст кнопки
 * @param onClick Callback при нажатии
 * @param modifier Modifier
 * @param size Размер кнопки (big, medium, small, xsmall)
 * @param enabled Активна ли кнопка
 */
@Composable
fun WfmLinkButton(
    text: String,
    onClick: () -> Unit,
    modifier: Modifier = Modifier,
    size: WfmLinkButtonSize = WfmLinkButtonSize.BIG,
    enabled: Boolean = true
) {
    val colors = WfmTheme.colors
    val textColor = if (enabled) colors.buttonLinkTextDefault else colors.buttonLinkTextDisabled

    val textStyle = when (size) {
        WfmLinkButtonSize.BIG -> WfmTypography.Headline16Medium
        WfmLinkButtonSize.MEDIUM -> WfmTypography.Headline14Medium
        WfmLinkButtonSize.SMALL, WfmLinkButtonSize.XSMALL -> WfmTypography.Headline12Medium
    }

    val (horizontalPadding, verticalPadding) = when (size) {
        WfmLinkButtonSize.BIG, WfmLinkButtonSize.MEDIUM -> WfmSpacing.L to WfmSpacing.M  // 16/12
        WfmLinkButtonSize.SMALL -> WfmSpacing.M to WfmSpacing.S  // 12/8
        WfmLinkButtonSize.XSMALL -> WfmSpacing.S to WfmSpacing.XXS  // 8/4
    }

    Box(
        modifier = modifier
            .then(
                if (enabled) {
                    Modifier.clickable(onClick = onClick)
                } else {
                    Modifier
                }
            )
            .padding(horizontal = horizontalPadding, vertical = verticalPadding),
        contentAlignment = Alignment.Center
    ) {
        Text(
            text = text,
            style = textStyle,
            color = textColor,
            textAlign = TextAlign.Center
        )
    }
}

/**
 * Размер Link кнопки
 */
enum class WfmLinkButtonSize {
    BIG,      // 16px font, 16/12 padding
    MEDIUM,   // 14px font, 16/12 padding
    SMALL,    // 12px font, 12/8 padding
    XSMALL    // 12px font, 8/4 padding
}

// MARK: - Preview

@Preview(name = "Link Button - All Sizes", showBackground = true)
@Composable
private fun WfmLinkButtonPreview() {
    WfmTheme {
        Column(
            modifier = Modifier.padding(16.dp),
            verticalArrangement = Arrangement.spacedBy(16.dp)
        ) {
            WfmLinkButton(
                text = "Отмена",
                size = WfmLinkButtonSize.BIG,
                onClick = {},
                modifier = Modifier.fillMaxWidth()
            )

            WfmLinkButton(
                text = "Отмена",
                size = WfmLinkButtonSize.MEDIUM,
                onClick = {},
                modifier = Modifier.fillMaxWidth()
            )

            WfmLinkButton(
                text = "Отмена",
                size = WfmLinkButtonSize.SMALL,
                onClick = {},
                modifier = Modifier.fillMaxWidth()
            )

            WfmLinkButton(
                text = "Отмена",
                size = WfmLinkButtonSize.XSMALL,
                onClick = {},
                modifier = Modifier.fillMaxWidth()
            )

            WfmLinkButton(
                text = "Отмена (Disabled)",
                enabled = false,
                onClick = {},
                modifier = Modifier.fillMaxWidth()
            )
        }
    }
}

File: wfm-develop mobile 07.05.2026/wfm/mobile/ios/WFMUI/Sources/WFMUI/Components/WFMLinkButton.swift
import SwiftUI

/// Link кнопка дизайн-системы WFM
///
/// Низкий приоритет, для менее важных действий.
/// Представляет собой текст без фона в цветах brand.
///
/// Дизайн: https://www.figma.com/design/J9crsNS4wIszpILzWvnY2T/Smart-employee?node-id=900-12300
public struct WFMLinkButton: View {
    let text: String
    let action: () -> Void
    let size: WFMLinkButtonSize
    let isEnabled: Bool

    @Environment(\.wfmColors) private var colors

    public init(
        text: String,
        size: WFMLinkButtonSize = .big,
        isEnabled: Bool = true,
        action: @escaping () -> Void
    ) {
        self.text = text
        self.size = size
        self.isEnabled = isEnabled
        self.action = action
    }

    public var body: some View {
        Button(action: {
            if isEnabled {
                action()
            }
        }) {
            buttonContent
        }
        .disabled(!isEnabled)
    }

    @ViewBuilder
    private var buttonContent: some View {
        Text(text)
            .font(font)
            .tracking(letterSpacing)
            .foregroundColor(textColor)
            .multilineTextAlignment(.center)
            .padding(.horizontal, horizontalPadding)
            .padding(.vertical, verticalPadding)
            .fixedSize(horizontal: false, vertical: true)
    }

    private var font: Font {
        switch size {
        case .big:
            return WFMTypography.headline16Medium
        case .medium:
            return WFMTypography.headline14Medium
        case .small:
            return WFMTypography.headline12Medium
        case .xsmall:
            return WFMTypography.headline12Medium
        }
    }

    private var letterSpacing: CGFloat {
        switch size {
        case .big:
            return WFMTypography.LetterSpacing.headline16Medium
        case .medium:
            return WFMTypography.LetterSpacing.headline14Medium
        case .small:
            return WFMTypography.LetterSpacing.headline12Medium
        case .xsmall:
            return WFMTypography.LetterSpacing.headline12Medium
        }
    }

    private var horizontalPadding: CGFloat {
        switch size {
        case .big, .medium:
            return WFMSpacing.l  // 16
        case .small:
            return WFMSpacing.m  // 12
        case .xsmall:
            return WFMSpacing.s  // 8
        }
    }

    private var verticalPadding: CGFloat {
        switch size {
        case .big, .medium:
            return WFMSpacing.m  // 12
        case .small:
            return WFMSpacing.s  // 8
        case .xsmall:
            return WFMSpacing.xxs  // 4
        }
    }

    private var textColor: Color {
        isEnabled ? colors.buttonLinkTextDefault : colors.buttonLinkTextDisabled
    }
}

/// Размер Link кнопки
public enum WFMLinkButtonSize {
    case big      // 16px font, 16/12 padding
    case medium   // 14px font, 16/12 padding
    case small    // 12px font, 12/8 padding
    case xsmall   // 12px font, 8/4 padding
}

// MARK: - Preview

#Preview("Link Button - All Sizes") {
    VStack(spacing: 16) {
        WFMLinkButton(text: "Отмена", size: .big) {
            print("Big button tapped")
        }

        WFMLinkButton(text: "Отмена", size: .medium) {
            print("Medium button tapped")
        }

        WFMLinkButton(text: "Отмена", size: .small) {
            print("Small button tapped")
        }

        WFMLinkButton(text: "Отмена", size: .xsmall) {
            print("XSmall button tapped")
        }

        WFMLinkButton(text: "Отмена (Disabled)", isEnabled: false) {
            print("Disabled button tapped")
        }
    }
    .padding()
    .wfmTheme()
}
